spec/index.adoc

```diff
@@ -556,18 +556,35 @@ If this `certificate` includes subnet delegations (possibly nested), then the `e
 - all requested paths have `/time` or `/request_status/<request_id>` as prefix where the (single) original request referenced by `<request_id>` is an update call to the Management Canister (`aaaaa-aa`) and the method name is `provisional_create_canister_with_cycles`, and
 - whenever the certificate contains the path `/request_status/<request_id>/reply`, then its value is a Candid-encoded record with a `canister_id` field of type `principal` and the `canister_id` must be included in each delegation’s canister id range.
 
-The returned certificate reveals all values whose path is a suffix of the list of requested paths. It also always reveals `/time`, even if not explicitly requested.
+The returned certificate reveals all values whose path is a suffix of a requested path. It also always reveals `/time`, even if not explicitly requested.
 
 All requested paths must have one of the following paths as prefix:
 
- * `/time`. Can be requested by anyone.
- * `/subnet`. Can be requested by anyone.
- * `/request_status/<request_id>`. Can only be requested by the same sender as the original request referenced by `<request_id>` and if the effective canister id of the original request matches `<effective_canister_id>`.
- * `/canisters/<canister_id>/module_hash`. Can be requested by anyone if `<canister_id>` matches `<effective_canister_id>`.
- * `/canisters/<canister_id>/controllers`. Can be requested by anyone if `<canister_id>` matches `<effective_canister_id>`. The order may vary depending on the implementation.
- * `/canisters/<canister_id>/metadata/<name>`. Can be requested by anyone if `<canister_id>` matches `<effective_canister_id>` and `<name>` is a public custom section. If `<name>` is a private custom section, it can only be requested by the controllers of the canister.
+ * `/time`. Can always be requested.
+ * `/subnet`. Can always be requested.
+ * `/request_status/<request_id>`. Can be requested if no path with such a prefix exists in the state tree or
+   - the sender of the original request referenced by `<request_id>` is the same as the sender of the read state request and
+   - `<effective_canister_id>` is a valid effective canister id for the original request referenced by `<request_id>` and
+   - the effective canister id of the original request referenced by `<request_id>` and the effective canister id `<effective_canister_id>` of the read state request belong to canister ranges of the same subnet.
```

Comment:
The condition "on the same subnet" seems at odds with the general approach of hiding the subnets where possible. Does it maybe make sense to _guarantee_ correct output if the `request_id` and `effective_request_id` are the same and not guarantee behavior otherwise?

Reply:
Now that I looked into the Rust implementation of this check (it doesn't seem possible to get the original message arguments from the ingress history which would be needed to check the second condition above), we might need to store the effective canister id of the original request in the ingress history and thus the original formulation (effective canister id of the original request must match the effective canister id of this read state request) might actually work.

Reply:
We might actually take the original condition (the effective canister id of the original request matches the effective canister id of this read state request) if we store the effective canister id of the original request in the ingress history.